tiledb/sm/filesystem/s3.cc

```diff
@@ -600,11 +600,11 @@ Status S3::ls(
   list_objects_request.SetDelimiter(delimiter.c_str());
   if (request_payer_ != Aws::S3::Model::RequestPayer::NOT_SET)
     list_objects_request.SetRequestPayer(request_payer_);
-  if (max_paths != -1)
-    list_objects_request.SetMaxKeys(max_paths);
 
   bool is_done = false;
   while (!is_done) {
+    if (max_paths != -1)
+      list_objects_request.SetMaxKeys(max_paths);
```

Comment:
Does this need to be set inside the while loop?

Reply:
@Shelnutt2 good catch! While testing another change I didn't revert this part properly. Look at the latest code, we set it inside the loop to limit the number of results in the last request in case we need more than one.